portal/static/js/src/CookieMonster.js

```diff
@@ -11,13 +12,28 @@ import {getUrlParameter} from "./modules/Utility.js";
     CookieMonster.prototype.deleteTestCookie = function() {
         var d = new Date();
         d.setTime(d.getTime() - (1000*60*60*24)); //set the time to the past. 1000 milliseonds = 1 second
-        var expires = "expires=" + d.toGMTString(); //compose the expirartion date
-        window.document.cookie = this.testCookieName+"="+"; "+expires;//set the cookie with name and the expiration date
+        //set the cookie with name and the expiration date
+        this.setCookie({"expires":d.toGMTString(), ...this.getSameSiteAttribute()});
+    };
+    CookieMonster.prototype.getSameSiteAttribute = function() {
+        //Do we need this set specifically to environment or ?
+        //if (String(this.settings["SYSTEM_TYPE"]).toLowerCase() === "development") return {};
```

Comment:
@ivan-c Ivan, in the story, you mentioned, `conditionally add Secure cookie attribute for non-development deploys`, do we not need it for other environments, e.g. development?

Reply:
> @ivan-c Ivan, in the story, you mentioned, `conditionally add Secure cookie attribute for non-development deploys`, do we not need it for other environments, e.g. development?

Correct. If a cookie has the `Secure` attribute, it is required to be transmitted over https only. We do not want to set the `Secure` attribute on cookies for development deployments, otherwise no cookies will be set, and you will not be able to login into your development deploy.

We have the same functionality in the [flask backend here](https://github.com/uwcirg/truenth-portal/blob/develop/portal/config/config.py#L185-L186), just need it replicated in the frontend.

Reply:
thank you, @ivan-c, for clarifying, I did just put the conditional check for environment back now.